mroderick/plete
# Fix issue 6 properly

This PR fixes a mistake I made in 23e0cea

#### Background

The fix applied in 23e0cea ended up breaking the filtering when `dataSrc` is an array of strings.

Given `["Denmark", "Germany", "Spain", "Sweden", "United Kingdom"]` it would not
return `["Denmark", "Sweden"]` for a query of `"Den"`.

#### Solution

Use a battle hardened escaping function, from https://stackoverflow.com/a/3561711

#### How to observe problem

1. Check out `9d9fe1781c22532832331c0fb93778e179e0f127` (current `master`)
1. `npm ci`
1. `npm start`
1. Navigate to http://localhost:8080/
1. Type "Den" into the input element of the "Simple input" example
1. Observe that there are no suggestions

![2020-01-22 at 13 28](https://user-images.githubusercontent.com/20321/72897979-10978c00-3d1b-11ea-8ff1-8347633f7b76.png)

#### How to verify solution

1. Check out this branch
1. `npm ci`
1. `npm start`
1. Navigate to http://localhost:8080/
1. Type "Den" into the input element of the "Simple input" example
1. Observe that there are two examples: "Denmark" and "Sweden"

![2020-01-22 at 13 26](https://user-images.githubusercontent.com/20321/72897943-ff4e7f80-3d1a-11ea-838a-5df5ccad2001.png)
